### DISCUSSION TITLE
Fetching partial entities with MongoDB does not work

### DISCUSSION THREAD

**Discussion by dyedwiper** [mikro-orm/mikro-orm#5599] [Q&A]
### Describe the bug

Fetching partial entities with MongoDB does not work for me. When I do a query like `await this.em.find(User, {  },  { fields: [ 'name' ] });`, I get the whole document and not just the selected field.

The logged query is `db.getCollection('user').find({}, { projection: { _id: 1, name: 1 } }).toArray();`. I'm not a MongoDB expert, but this seems wrong syntax. It has to be `db.getCollection('someName').find({},  { _id: 1, name: 1 }).toArray();`. 

See: https://www.mongodb.com/docs/manual/reference/method/db.collection.find/#projections
Here is an example in a MongoDB playground: https://www.humongous.io/app/playground/mongodb/66509ca29052cdbd8bfca188

I guess this line of code has to do with it: https://github.com/mikro-orm/mikro-orm/blob/master/packages/mongodb/src/MongoConnection.ts#L173

### Reproduction

I tried to reproduce a minimal example but it somehow does not run in CodeSandBox: https://codesandbox.io/p/devbox/relaxed-fermi-5c8294

I can execute it locally and weirdly enough there it works. Maybe it has something to with the particular in-memory-db that is used? But like I said: I'm not an expert in MongoDB.

### What driver are you using?

@mikro-orm/mongodb

### MikroORM version

5.6.16

### Node.js version

18.14.0

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Upgrade, v5 is no longer supported (and you are on an old v5 version too, at least try latest v5). Also providing a repro that is not reproducing anything wont help much. 

My guess is your entity is already loaded in the context, try adding refresh: true to the find options, if its about that, it should help.

**Comment by dyedwiper** [comment]
Your guess was right. I was hasty there. Sorry! 

But I'm still confused why the logged query does not work when I directly execute it in MongoDB.
